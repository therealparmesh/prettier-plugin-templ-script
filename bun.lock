{
  "lockfileVersion": 1,
  "workspaces": {
    "": {
      "name": "prettier-plugin-templ-script",
      "devDependencies": {
        "prettier": "^3.0.0",
        "prettier-plugin-tailwindcss": "^0.6.0",
      },
      "peerDependencies": {
        "prettier": "*",
      },
    },
  },
  "packages": {
    "prettier": ["prettier@3.6.2", "", { "bin": { "prettier": "bin/prettier.cjs" } }, "sha512-I7AIg5boAr5R0FFtJ6rCfD+LFsWHp81dolrFD8S79U9tb8Az2nGrJncnMSnys+bpQJfRUzqs9hnA81OAA3hCuQ=="],

    "prettier-plugin-tailwindcss": ["prettier-plugin-tailwindcss@0.6.13", "", { "peerDependencies": { "@ianvs/prettier-plugin-sort-imports": "*", "@prettier/plugin-pug": "*", "@shopify/prettier-plugin-liquid": "*", "@trivago/prettier-plugin-sort-imports": "*", "@zackad/prettier-plugin-twig": "*", "prettier": "^3.0", "prettier-plugin-astro": "*", "prettier-plugin-css-order": "*", "prettier-plugin-import-sort": "*", "prettier-plugin-jsdoc": "*", "prettier-plugin-marko": "*", "prettier-plugin-multiline-arrays": "*", "prettier-plugin-organize-attributes": "*", "prettier-plugin-organize-imports": "*", "prettier-plugin-sort-imports": "*", "prettier-plugin-style-order": "*", "prettier-plugin-svelte": "*" }, "optionalPeers": ["@ianvs/prettier-plugin-sort-imports", "@prettier/plugin-pug", "@shopify/prettier-plugin-liquid", "@trivago/prettier-plugin-sort-imports", "@zackad/prettier-plugin-twig", "prettier-plugin-astro", "prettier-plugin-css-order", "prettier-plugin-import-sort", "prettier-plugin-jsdoc", "prettier-plugin-marko", "prettier-plugin-multiline-arrays", "prettier-plugin-organize-attributes", "prettier-plugin-organize-imports", "prettier-plugin-sort-imports", "prettier-plugin-style-order", "prettier-plugin-svelte"] }, "sha512-uQ0asli1+ic8xrrSmIOaElDu0FacR4x69GynTh2oZjFY10JUt6EEumTQl5tB4fMeD6I1naKd+4rXQQ7esT2i1g=="],
  }
}
