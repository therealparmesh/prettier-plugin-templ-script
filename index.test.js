import { expect, test } from 'bun:test';
import prettier from 'prettier';
import * as templPlugin from '.';

function formatTempl(input, options = {}) {
  return prettier.format(input, {
    parser: 'templ',
    plugins: [templPlugin, 'prettier-plugin-tailwindcss'],
    singleQuote: true,
    ...options,
  });
}

test('should format JavaScript in script elements', async () => {
  const input = `<div>
	<script>
		(()=>{console.log("Hello world");})();
	</script>
</div>`;
  const result = await formatTempl(input);
  expect(result).toMatchSnapshot();
});

test('should format class attributes (Tailwind CSS ordering example)', async () => {
  const input = `<div class="hover:bg-red-600 bg-red-500 text-white p-4">
	Content
</div>`;
  const result = await formatTempl(input);
  expect(result).toMatchSnapshot();
});

test('should format JavaScript in script elements and class attributes (Tailwind CSS ordering example)', async () => {
  const input = `<div class="hover:bg-blue-600 bg-blue-500 text-white p-4">
	<script>
		(()=>{console.log("Combined test");const x=1;if(x>0){console.log("Hello world");}})();
	</script>
</div>`;
  const result = await formatTempl(input);
  expect(result).toMatchSnapshot();
});

test('should handle multiple script elements', async () => {
  const input = `<div>
	<script>
		(()=>{console.log("First script");})();
	</script>
	<script>
		(()=>{console.log("Second script");})();
	</script>
</div>`;
  const result = await formatTempl(input);
  expect(result).toMatchSnapshot();
});

test('should handle empty script tag', async () => {
  const input = `<div>
	<script>
	</script>
</div>`;
  const result = await formatTempl(input);
  expect(result).toMatchSnapshot();
});

test('should format JavaScript in script elements with attributes', async () => {
  const input = `<div>
	<script type="module">
		import{log}from"./utils.js";log("Hello from module");const data={x:1,y:2};console.log(data);
	</script>
</div>`;
  const result = await formatTempl(input);
  expect(result).toMatchSnapshot();
});

test('should format JSON in script elements with type="importmap"', async () => {
  const input = `<div>
	<script type="importmap">
		{"imports":{"react":"https://cdn.skypack.dev/react","lodash":"https://cdn.skypack.dev/lodash"}}
	</script>
</div>`;
  const result = await formatTempl(input);
  expect(result).toMatchSnapshot();
});

test('should format both script elements and class attributes when templMode is both', async () => {
  const input = `<div class="hover:bg-green-600 bg-green-500 text-white p-4">
	<script>
	  (()=>{console.log("Hello world");})();
	</script>
</div>`;
  const result = await formatTempl(input, { templMode: 'both' });
  expect(result).toMatchSnapshot();
});

test('should only format script elements when templMode is script-only', async () => {
  const input = `<div class="hover:bg-red-600 bg-red-500 text-white p-4">
	<script>
	  (()=>{console.log("Hello world");})();
	</script>
</div>`;
  const result = await formatTempl(input, { templMode: 'script-only' });
  expect(result).toMatchSnapshot();
});

test('should only format class attributes when templMode is class-only', async () => {
  const input = `<div class="hover:bg-blue-600 bg-blue-500 text-white p-4">
	<script>
	  (()=>{console.log("Hello world");})();
	</script>
</div>`;
  const result = await formatTempl(input, { templMode: 'class-only' });
  expect(result).toMatchSnapshot();
});
