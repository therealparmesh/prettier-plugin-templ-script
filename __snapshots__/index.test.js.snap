// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`should format JavaScript in script elements 1`] = `
"<div>
	<script>
		(() => {
			console.log('Hello world');
		})();
	</script>
</div>"
`;

exports[`should format class attributes (Tailwind CSS ordering example) 1`] = `
"<div class="bg-red-500 p-4 text-white hover:bg-red-600">
	Content
</div>"
`;

exports[`should format JavaScript in script elements and class attributes (Tailwind CSS ordering example) 1`] = `
"<div class="bg-blue-500 p-4 text-white hover:bg-blue-600">
	<script>
		(() => {
			console.log('Combined test');
			const x = 1;
			if (x > 0) {
				console.log('Hello world');
			}
		})();
	</script>
</div>"
`;

exports[`should handle multiple script elements 1`] = `
"<div>
	<script>
		(() => {
			console.log('First script');
		})();
	</script>
	<script>
		(() => {
			console.log('Second script');
		})();
	</script>
</div>"
`;

exports[`should handle empty script tag 1`] = `
"<div>
	<script>
	</script>
</div>"
`;

exports[`should format JavaScript in script elements with attributes 1`] = `
"<div>
	<script type="module">
		import { log } from './utils.js';
		log('Hello from module');
		const data = { x: 1, y: 2 };
		console.log(data);
	</script>
</div>"
`;

exports[`should format JSON in script elements with type="importmap" 1`] = `
"<div>
	<script type="importmap">
		{
			"imports": {
				"react": "https://cdn.skypack.dev/react",
				"lodash": "https://cdn.skypack.dev/lodash"
			}
		}
	</script>
</div>"
`;

exports[`should format both script elements and class attributes when templMode is both 1`] = `
"<div class="bg-green-500 p-4 text-white hover:bg-green-600">
	<script>
		(() => {
			console.log('Hello world');
		})();
	</script>
</div>"
`;

exports[`should only format script elements when templMode is script-only 1`] = `
"<div class="hover:bg-red-600 bg-red-500 text-white p-4">
	<script>
		(() => {
			console.log('Hello world');
		})();
	</script>
</div>"
`;

exports[`should only format class attributes when templMode is class-only 1`] = `
"<div class="bg-blue-500 p-4 text-white hover:bg-blue-600">
	<script>
	  (()=>{console.log("Hello world");})();
	</script>
</div>"
`;
